hrreport/CRM/HRReport/Form/Activity/HRAbsenceDates.php

```diff
@@ -0,0 +1,640 @@
+          'grouping' => 'contact-fields',
+        ],
+      'civicrm_email' =>
+        [
+          'dao' => 'CRM_Core_DAO_Email',
+          'fields' =>
+            [
+              'contact_target_email' =>
+                [
+                  'name' => 'email',
+                  'title' => ts('Email'),
+                  'alias' => 'civicrm_email_target',
+                  'no_display' => TRUE,
+                ],
+              'contact_assignee_email' =>
+                [
+                  'name' => 'email',
+                  'title' => ts('Approved By (Email)'),
+                  'alias' => 'civicrm_email_assignee',
+                  'no_display' => TRUE,
+                ],
+              'contact_source_email' =>
+                [
+                  'name' => 'email',
+                  'title' => ts('Added By (Email)'),
+                  'alias' => 'civicrm_email_source',
+                  'no_display' => TRUE,
+                ],
+            ]
+        ],
+      'civicrm_activity' =>
+        [
+          'dao' => 'CRM_Activity_DAO_Activity',
+          'fields' =>
+            [
+              'id' =>
+                [
+                  'no_display' => TRUE,
+                  'title' => ts('Activity ID'),
+                  'required' => TRUE,
+                ],
+              'source_record_id' =>
+                [
+                  'title' => ts('Absence ID'),
+                  'required' => TRUE
+                ],
+              'activity_type_id' =>
+                [
+                  'title' => ts('Absence Type'),
+                  'required' => TRUE,
+                  'type' => CRM_Utils_Type::T_STRING,
+                ],
+              'activity_subject' =>
+                [
+                  'title' => ts('Subject'),
+                  'default' => TRUE,
+                  'no_display' => TRUE,
+                ],
+              'activity_date_time' =>
+                [
+                  'title' => ts('Absence Date'),
+                  'required' => TRUE,
+                ],
+              'duration' =>
+                [
+                  'title' => ts('Qty'),
+                  'required' => TRUE,
+                ],
+              'status_id' =>
+                [
+                  'title' => ts('Status'),
+                  'default' => TRUE,
+                  'type' => CRM_Utils_Type::T_STRING,
+                  'required' => TRUE,
+                ],
+              'details' =>
+                [
+                  'title' => ts('Comments'),
+                  'default' => TRUE,
+                  'type' => CRM_Utils_Type::T_STRING,
+                  'required' => TRUE,
+                ],
+            ],
+          'filters' =>
+            [
+              'activity_type_id' =>
+                [
+                  'title' => ts('Absence Type'),
+                  'operatorType' => CRM_Report_Form::OP_MULTISELECT,
+                  'options' => $this->absenceActivityType,
```

Comment:
The child activity don't hold the absence type information and all of them are of type `absence`.  The absence type information was added in the `alterDisplay` function after the query might have ran and the rows fetched. 
Since the parent activities were not part of the original query, it will be difficult to filter by the various absence types and I did not want to waste too much time on that. I can remove this filter because even if I ad back the absence types list, it won't work.
What do you suggest?